docs/source/stability_policy.rst

```diff
@@ -0,0 +1,137 @@
+----------
+
+PTB uses a versioning scheme that roughly follows `https://semver.org/ <https://semver.org/>`_, although it may not be quite as strict.
+
+Given a version of PTB X.Y.Z,
+
+-  X indicates the major version number.
+   This is incremented when backwards incompatible changes are introduced.
+-  Y indicates the minor version number.
+   This is incremented when new functionality or backwards compatible changes are introduced by PTB.
+   *This is also incremented when PTB adds support for a new Bot API version, which may include backwards incompatible changes in some cases.*
+-  Z is the patch version.
+   This is incremented for backwards incompatible changes or other major changes.
+
+Deprecation
+~~~~~~~~~~~
+
+From time to time we will want to change the behavior of an API or remove it entirely, or we do so to comply with changes in the Telegram Bot API.
+In those cases, we follow a deprecation schedule as detailed below.
+
+Functionality is marked as deprecated by a corresponding note in the release notes and the documentation.
+Were possible, a :class:`~telegram.warnings.PTBDeprecationWarning` is issued when deprecated functionality is used, but this is not mandatory.
+
+From time to time, we may decide to deprecate an API that is particularly widely used.
+In these cases, we may decide to provide an extended deprecation period, at our discretion.
+
+With version 20.0.0, PTB introduced majorly structural breaking changes without the above deprecation period.
+Should a similarly big change ever be deemed necessary by the development team and should a deprecation period prove too much additional effort, this violation of the stability policy will be announced well ahead of the release in our channel.
+
+.. _non-bot-api-functionality-1:
+
+Non-Bot API Functionality
+#########################
+
+Starting with version NEXT.VERSION, deprecated functionality will stay available for the current and the next major version.
+More precisely:
+
+-  In PTB X.Y.Z the feature exists
+-  In PTB X.Y.(Z + 1) or X.(Y + 1).\* the feature is marked as deprecated
+-  In PTB (X + 1).\ *.* the feature is marked as deprecated
+-  In PTB (X + 2).0.0 the feature is removed or changed
+
+.. _bot-api-functionality-2:
+
+Bot API Functionality
+#####################
+
+As PTB has no control over deprecations introduced by Telegram and the schedule of these deprecations rarely coincides with PTBs deprecation schedule, we have a special policy for Bot API functionality.
+
+Starting with NEXT.VERSION, deprecated Bot API functionality will stay available for the current and the next major version *or* until the next version of the Bot API.
+More precisely, two cases are possible.
+
+Case 1
+^^^^^^
+
+-  In PTB X.Y.Z the feature exists
+-  Bot API version N.M is released and deprecates the feature
+-  In PTB X.(Y + 1).0 adds support for Bot API A.B and the feature is
+   marked as deprecated
+-  In PTB (X + 1).0.0 the feature is removed or changed
+
+Case 2
```

Comment:
Case 2 doesn't make sense to me since the bot api has never removed a feature totally (other than 64 bit user ids), so this has never happened right?

Reply:
I got more confused :( Is there a specific example in the commit history? When did we drop backward-compatibility-hacks like this? Would rewriting an already "patched" bot method to include new parameters count as an example?

Reply:
I don't recall that we already had such a case in the history, no. I guess we so far always made breaking changes in such cases. In the API 6.6 case my example would be that once API 6.7 is released, I want to drop all the `thumb(nail)` workarounds.

I converted the schematic overview to examples :)